blockmanager.go

```diff
@@ -292,13 +294,53 @@ type headerNode struct {
 	hash   *chainhash.Hash
 }
 
+// PeerNotifier provides an interface for server peer notifications.
+type PeerNotifier interface {
+	// AnnounceNewTransactions generates and relays inventory vectors and
+	// notifies both websocket and getblocktemplate long poll clients of
+	// the passed transactions.
+	AnnounceNewTransactions(txns []*dcrutil.Tx)
+
+	// UpdatePeerHeights updates the heights of all peers who have have
+	// announced the latest connected main chain block, or a recognized orphan.
+	UpdatePeerHeights(latestBlkHash *chainhash.Hash, latestHeight int64, updateSource *serverPeer)
+
+	// RelayInventory relays the passed inventory vector to all connected peers
+	// that are not already known to have it.
+	RelayInventory(invVect *wire.InvVect, data interface{}, immediate bool)
+
+	// TransactionConfirmed marks the provided single confirmation transaction
+	// as no longer needing rebroadcasting.
+	TransactionConfirmed(tx *dcrutil.Tx)
+}
+
+// blockManangerConfig is a configuration struct for a blockManager.
+type blockManagerConfig struct {
+	Ctx          context.Context
+	PeerNotifier PeerNotifier
+	TimeSource   blockchain.MedianTimeSource
+
+	// The following fields are for accessin the chain and its configuration.
+	Chain       *blockchain.BlockChain
+	ChainParams *chaincfg.Params
+
+	// The following fields provide access to the fee estimator, mempool and
+	// the background block template generator.
+	FeeEstimator       *fees.Estimator
+	TxMemPool          *mempool.TxPool
+	BgBlkTmplGenerator *BgBlkTmplGenerator
+
+	// The following fields are blockManger callbacks.
+	NotifyWinningTickets      func(*WinningTicketsNtfnData)
+	PruneRebroadcastInventory func()
+}
+
 // blockManager provides a concurrency safe block manager for handling all
 // incoming blocks.
 type blockManager struct {
-	server          *server
+	cfg             *blockManagerConfig
 	started         int32
 	shutdown        int32
-	chain           *blockchain.BlockChain
```

Comment:
I would prefer this to be left and set when creating the instance from the config in order to significantly minimize the overall diff.  It would remove the need for every change from `b.chain` -> `b.cfg.Chain`.